### PR TITLE
Results.stream overaloads /2

### PR DESCRIPTION
**Pull request by Rick-Anderson**
Contributes to #26413
[Internal review URL](https://review.docs.microsoft.com/en-us/aspnet/core/fundamentals/minimal-apis?view=aspnetcore-7.0&branch=pr-en-us-26412#stream-1)

## Public review  of the doc build:
- download the [HTML files](https://www.dropbox.com/sh/yy994551cje0kap/AAB0tRnm7gze2kYrSYPSFHIda?dl=0)
- unzip
- view HTML file in a browser. It best to view the doc with the widest window that has no left or right TOC (Table  of contents)
- For links outside the doc, remove the `review.` from URL.

### CODE REVIEW

**Comment by Rick-Anderson** [comment]
@fiyazbinhasan  your samples are really helpful. Please review the public document and take a quick look at your samples I modified.

**Comment by fiyazbinhasan** [comment]
@Rick-Anderson the downloadable HTML files are for Minimal API filters

**Comment by Rick-Anderson** [comment]
> @Rick-Anderson the downloadable HTML files are for Minimal API filters

Yes, that's it. Look for SixLabors

**Comment by Rick-Anderson** [comment]
@halter73  who should do a quick review of 
[Internal review URL](https://review.docs.microsoft.com/en-us/aspnet/core/fundamentals/minimal-apis?view=aspnetcore-7.0&branch=pr-en-us-26412#stream-1)
[Results.Stream](https://review.docs.microsoft.com/en-us/dotnet/api/microsoft.aspnetcore.http.results.stream?view=aspnetcore-7.0&preserve-view=true) overloads allow access to the underlying HTTP response stream without buffering.

File: aspnetcore/fundamentals/minimal-apis/resultsStream/7.0-samples/ResultsStreamSample/Program.cs [hunk 1]
```diff
@@ -48,14 +76,86 @@ async Task ProcessImage(Stream stream, CancellationToken token)
...
+    BlobContainerClient blobContainerClient = new BlobContainerClient(conStr, containerName);
+
+    if (!blobContainerClient.Exists())
+    {
+        blobContainerClient.Create();
+    }
+
+    BlobClient blob = blobContainerClient.GetBlobClient(blobName);
+
+    blob.Upload($"wwwroot/img/{blobName}");
```
Comment (halter73) [inline aspnetcore/fundamentals/minimal-apis/resultsStream/7.0-samples/ResultsStreamSample/Program.cs:112]: Shouldn't this be async?

File: aspnetcore/fundamentals/minimal-apis/resultsStream/7.0-samples/ResultsStreamSample/Program.cs [hunk 2]
```diff
@@ -48,14 +76,86 @@ async Task ProcessImage(Stream stream, CancellationToken token)
...
+{
+    var conStr = builder.Configuration["blogConStr"];
+    BlobServiceClient blobServiceClient = new BlobServiceClient(conStr);
+
+    await ListContainers(blobServiceClient, "",99,app.Logger);
+});
 
 app.Run();
+
+async static Task ListContainers(BlobServiceClient blobServiceClient,
```
Comment (halter73) [inline aspnetcore/fundamentals/minimal-apis/resultsStream/7.0-samples/ResultsStreamSample/Program.cs:137]:
```suggestion
async static Task ListContainersAsync(BlobServiceClient blobServiceClient,
```

File: aspnetcore/fundamentals/minimal-apis/resultsStream/7.0-samples/ResultsStreamSample/Program.cs [hunk 3]
```diff
@@ -48,14 +76,86 @@ async Task ProcessImage(Stream stream, CancellationToken token)
...
+    }
+});
+
+// GET 
+app.MapGet("/list-containers/", async () =>
+{
+    var conStr = builder.Configuration["blogConStr"];
+    BlobServiceClient blobServiceClient = new BlobServiceClient(conStr);
+
+    await ListContainers(blobServiceClient, "",99,app.Logger);
```
Comment (halter73) [inline aspnetcore/fundamentals/minimal-apis/resultsStream/7.0-samples/ResultsStreamSample/Program.cs:132]:
```suggestion
    await ListContainers(blobServiceClient, "", 99, app.Logger);
```

Should we also just make this the root (`app.MapGet("/", ...`)?

File: aspnetcore/fundamentals/minimal-apis/resultsStream/7.0-samples/ResultsStreamSample/Program.cs [hunk 4]
```diff
@@ -8,35 +10,61 @@
...
+    int width = image.Width / 2;
+    int height = image.Height / 2;
+    image.Mutate(x => x.Rotate(RotateMode.Rotate90)
+    );
+    await image.SaveAsync(stream, JpegFormat.Instance, cancellationToken: token);
+}
+
+// GET /stream-image/microsoft.jpg/pictures
+// <snippet_abs>
+app.MapGet("/stream-image/{blobName}/{containerName}",
```
Comment (halter73) [inline aspnetcore/fundamentals/minimal-apis/resultsStream/7.0-samples/ResultsStreamSample/Program.cs:50]:
```suggestion
app.MapGet("/stream-image/{containerName}/{blobName}", 
```

**Review by halter73 — COMMENTED** [review]
`containerName` should probably go before the `blobName` in all routes. There's no technical reason, but I think putting the blob name before the container name is an unusual choice.

Also, we should probably stick to ending async method names and `Async`, so

- `ListContainers` -> `ListContainersAsync`
- `Resize` -> `ResizeAsync`/`ResizeImageAsync`
- `RotateImage` -> `RotateAsync`/`RotateImageAsync`

**Review by halter73 — APPROVED** [review]

**Pull request merged**